Source/Core/Logging/FramesEventLogger.swift

```diff
@@ -14,24 +14,27 @@ protocol FramesEventLogging {
 
 final class FramesEventLogger: FramesEventLogging {
 
-    private let correlationID: String
+    private let getCorrelationID: () -> String
     private let checkoutEventLogger: CheckoutEventLogging
     private let dateProvider: DateProviding
     
     // MARK: - Init
     
-    init(correlationID: String,
+    init(getCorrelationID: @escaping () -> String,
          checkoutEventLogger: CheckoutEventLogging,
          dateProvider: DateProviding) {
-      self.correlationID = correlationID
+      self.getCorrelationID = getCorrelationID
       self.checkoutEventLogger = checkoutEventLogger
       self.dateProvider = dateProvider
     }
 
     // MARK: - FramesEventLogging
     
     func log(_ framesLogEvent: FramesLogEvent) {
-        
+
+        // by setting correlationID for every log, we always keep in sync with Checkout SDK.
+        add(metadata: getCorrelationID(), forKey: .correlationID)
```

Comment:
Do we expect that correlation ID to change since initialised over the lifetime of the SDK?

If so, would it be an option to change the type on the source of the variable (from what I see `Checkout. checkoutAPIService .correlationID`), and have it be of type `NSMutableString`?

Whilst `String` is a value type, and `NSString` is a reference type binded to a value (behaves like value type), `NSMutableString` can actually act as a mutable reference type. This can save the passing a closure from place to place and simplifying the chain. Biggest drawback though, the source of the information should have its value set via `.setString` instead of `=`, for the behaviour to work.

Example:
```
var str: NSMutableString = NSMutableString(string: "OLD")
var newStr = str
newStr.setString("NEW")
print(str, newStr) // New, New

str = "Other change"
print(str, newStr) // Other change, New

```

Reply:
This would be a good feature add in Checkout SDK, then it can be implemented here

Reply:
@harry-brown-cko can you create a tech investment ticket for this which can go into checkout SDK.